### DISCUSSION TITLE
[Support] npm install @midwayjs/cli-plugin-swc 之后报错 MidwayConfigMissingError: Can't found config key "config.keys" in your config, please set it first

### DISCUSSION THREAD

**Discussion by Fc-404** [midwayjs/midway#3782] [Q&A]
### Describe the problem（描述问题）

我的Midway项目ts文件比较多，有80+个，所以我按照文档，在原有dev命令上添加了swc选项：
```` json
"dev": "cross-env TS_NODE_TYPE_CHECK=false TS_NODE_TRANSPILE_ONLY=true NODE_ENV=local midway-bin dev --ts --fast=swc",
````
然后安装了 @midwayjs/cli-plugin-swc 库
```` shell
npm install -D @midwayjs/cli-plugin-swc
````
但是我运行 npm run dev 就会报错
```` shell
start error: Can't found config key "config.keys" in your config, please set it first
MidwayConfigMissingError: Can't found config key "config.keys" in your config, please set it first
    at MidwayKoaFramework.applicationInitialize (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\koa\dist\framework.js:39:19)
    at MidwayKoaFramework.initialize (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\core\dist\baseFramework.js:61:20)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at MidwayFrameworkService.init (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\core\dist\service\frameworkService.js:99:21)
    at ObjectCreator.doInitAsync (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\core\dist\definitions\objectCreator.js:111:17)
    at ManagedResolverFactory.createAsync (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\core\dist\context\managedResolverFactory.js:258:9)
    at initializeGlobalApplicationContext (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\core\dist\setup.js:37:5)
    at create (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\mock\dist\creator.js:144:9)
    at createApp (C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\mock\dist\creator.js:180:23)
    at C:\Users\xazht\Documents\GitHub\XAZH\xazh-back\node_modules\@midwayjs\cli-plugin-dev\src\child.ts:52:13 {
  code: 'MIDWAY_10006',
  cause: undefined
}
````
我能确保我存在keys值，在我的config.default.ts中有
```` ts
import { MidwayConfig } from '@midwayjs/core';

export default {
  // use for cookie sign key, should change to your own and keep security
  keys: '2601721443@qq.com',
  koa: {
    port: 7001,
  },
} as MidwayConfig;
````
我尝试断点到 @midwayjs\koa\dist\framework.js:39 ，发现
```` js
    async applicationInitialize(options) {
        const appKeys = this.configService.getConfiguration('keys') ||
            this.configurationOptions['keys'];
        if (!appKeys) {
            throw new core_1.MidwayConfigMissingError('config.keys');
        }
        ......
    }
````
这里 this.configService.getConfiguration("keys") 获取不到 keys 值，我监视变量 this.configService，发现该对象下的configuration对象中的 keys 值是空的，但是default对象的keys是我config.default.ts中的值。

![image](https://github.com/midwayjs/midway/assets/33905772/122c86a1-9582-47fd-ac82-358bf5cd1a4b)

### Midway Versions（Midway 版本）

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(3.15.8)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.15.8)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.15.8)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(3.15.8)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.15.8)
✓ @midwayjs/info(3.15.8)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.15.8)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(not installed)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(3.15.8)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.15.8)
✓ @midwayjs/version(3.15.8)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.15.8)
✓ @midwayjs/ws(not installed)

**Comment by czy88840616** [comment]
用新的 mwtsc 试试？
